doc/classes/CanvasItem.xml

```diff
@@ -622,7 +622,7 @@
 			The rendering layer in which this [CanvasItem] is rendered by [Viewport] nodes. A [Viewport] will render a [CanvasItem] if it and all its parents share a layer with the [Viewport]'s canvas cull mask.
 		</member>
 		<member name="visible" type="bool" setter="set_visible" getter="is_visible" default="true">
-			If [code]true[/code], this [CanvasItem] is drawn. The node is only visible if all of its ancestors are visible as well (in other words, [method is_visible_in_tree] must return [code]true[/code]).
+			If [code]true[/code], this [CanvasItem] may be drawn. Whether this [CanvasItem] is actually drawn depends on the visibility of all of its [CanvasItem] ancestors. In other words: [method is_visible_in_tree] must return [code]true[/code] for this [CanvasItem] to be drawn.
```

Comment:
Visibility of not only CanvasItem ancestors matter, visibility of parent CanvasLayer/Window also might matter. There's also case when `top_level = true`. Also `visibility_layer` of this/parent items and relevant Viewport.

Just saying, no idea if/how exactly you want to clarify all of that in here. :upside_down_face:

Reply:
Hmmm, this is getting quite elaborate, haha. Perhaps I should refer more specifically to visible_in_tree in all the places where a short description of this fits best, and elaborate more on this in the description of that method only.

Reply:
Actually nevermind the `top_level = true` case, seems it doesn't affect visibility propagation, it still takes parent items' visibility into account regardless of the fact that transforms are not taken into account.

Note that `visible_in_tree` doesn't account for `visibility_layer`s though, so even if it returns `true` the item might end up not being rendered.

Reply:
Maybe a general paragraph about visibility could be added to the class description. But I don't think we could link to it, can we?

Reply:
I just checked and toggling the visibility layer doesn't emit the `visibility_changed` and `hidden` signal.

Reply:
Indeed it does not. One thing is whether the CanvasItem _wants to be_ rendered, and the other is whether it _can_ be rendered based on external factors.

Reply:
Added remarks about the `visibility_layer` to member `visible` and method `is_visible_in_tree`. Please review!